tools/syz-trace2syz/proggen/generate_unions.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2018 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package proggen
+
+import (
+	"strings"
+
+	"github.com/google/syzkaller/pkg/log"
+	"github.com/google/syzkaller/prog"
+	"github.com/google/syzkaller/tools/syz-trace2syz/parser"
+)
+
+func genIpv4Addr(syzType *prog.UnionType, irType parser.IrType, ctx *Context) prog.Arg {
+	if syzType.Dir() == prog.DirOut {
```

Comment:
This part seems to be duplicated in multiple places. Do we want to move it to genArgs? When does it make sense to not generate default for output arg?

Reply:
still relevant

Reply:
I think that we can move this up to genUnion(). In general I don't think we can move this to genArgs because as I mentioned earlier, there are cases where resources can be produced in an output Arg (pipe, socketpair, etc).